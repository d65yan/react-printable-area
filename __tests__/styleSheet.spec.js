import getStyles from '../src/styleSheet';

describe('', () => {
  const styles = getStyles('customClassName');

  it('uses the given class to set the selectors', () => {
    const screenSelector = `@media screen {
        body .customClassName {`;

    const printSelector = `@media print {
        body>*:not(.customClassName) {`;

    expect(styles).toContain(screenSelector);
    expect(styles).toContain(printSelector);
  });

  it('styles regression', () => {
    expect(styles).toMatchSnapshot();
  });
});
