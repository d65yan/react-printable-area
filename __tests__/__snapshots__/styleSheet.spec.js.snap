// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styles regression 1`] = `
"
    <style id=\\"print-styles-for-customClassName\\">
      @media screen {
        body .customClassName {
          display: block;
          width: 0;
          height: 0;
          overflow: hidden;
        }
      }


      @media print {
        body>*:not(.customClassName) {
          display: none !important;
        }
        body .customClassName {
          width: 100%;
        }
      }
    </style>
  "
`;
